cmocka: can't build make check for host target.

make check fails for host target.
```
./scripts/host-build-all.sh
....
```
then run
```
make check
...
make[3]: Entering directory '/home/lrg/source/reef/sof.git/test/cmocka'
make  mixer  preproc_varargs_count preproc_concat preproc_seq preproc_defer preproc_get_arg debugability_macros rstrcmp rstrlen strcheck volume_process buffer_new buffer_write buffer_wrap buffer_copy comp_set_state list_init list_is_empty list_item_append list_item_del list_item_is_last list_item_prepend list_item gcd ceil_divide find_equal_int16 find_min_int16 find_max_abs_int32 norm_int32 sin_fixed
make[4]: Entering directory '/home/lrg/source/reef/sof.git/test/cmocka'
depbase=`echo src/audio/mixer/mixer_test.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I. -I../../src/include    -O3 -Wall -Werror -Wl,-EL -Wmissing-prototypes -Wimplicit-fallthrough=3 -I./include -I../../src/arch/host/include -I /home/lrg/source/reef/sof.git/src/include -g -O2 -MT src/audio/mixer/mixer_test.o -MD -MP -MF $depbase.Tpo -c -o src/audio/mixer/mixer_test.o src/audio/mixer/mixer_test.c &&\
mv -f $depbase.Tpo $depbase.Po
In file included from ../../src/include/sof/trace.h:39,
                 from ../../src/include/sof/ipc.h:36,
                 from src/audio/mixer/mixer_test.c:38:
../../src/include/sof/mailbox.h:35:10: fatal error: platform/mailbox.h: No such file or directory
 #include <platform/mailbox.h>
          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[4]: *** [Makefile:1382: src/audio/mixer/mixer_test.o] Error 1
make[4]: Leaving directory '/home/lrg/source/reef/sof.git/test/cmocka'
make[3]: *** [Makefile:1900: check-am] Error 2
make[3]: Leaving directory '/home/lrg/source/reef/sof.git/test/cmocka'
make[2]: *** [Makefile:1903: check] Error 2
make[2]: Leaving directory '/home/lrg/source/reef/sof.git/test/cmocka'
make[1]: *** [Makefile:382: check-recursive] Error 1
make[1]: Leaving directory '/home/lrg/source/reef/sof.git/test'
make: *** [Makefile:433: check-recursive] Error 1
```

--- a/src/library/include/platform/platform.h
+++ b/src/library/include/platform/platform.h
@@ -55,8 +55,8 @@
 #define HOST_PAGE_SIZE		4096
 
 /* Platform stream capabilities */
-#define PLATFORM_MAX_CHANNELS	4
-#define PLATFORM_MAX_STREAMS	5
+#define PLATFORM_MAX_CHANNELS	8
+#define PLATFORM_MAX_STREAMS	16
 
 /* DMA channel drain timeout in microseconds */
 #define PLATFORM_DMA_TIMEOUT	1333

--- a/test/cmocka/Makefile.am
+++ b/test/cmocka/Makefile.am
@@ -116,8 +116,10 @@ preproc_get_arg_SOURCES =\
 
 # debugability tests
 
+if BUILD_XTENSA
 check_PROGRAMS += debugability_macros
 debugability_macros_SOURCES = src/debugability/macros.c
+endif
 
 # lib/lib tests
 
@@ -150,6 +152,9 @@ endif
 
 # volume tests
 
+if BUILD_XTENSA
+# these tests should be able to run on host after some tuning
+
 check_PROGRAMS += volume_process
 volume_process_SOURCES = src/audio/volume/volume_process.c
 volume_process_CFLAGS = -I../../src/audio $(AM_CFLAGS)
@@ -165,7 +170,12 @@ else
 volume_process_LDADD =  ../../src/audio/libaudio.a $(LDADD)
 endif
 
+endif
+
 # buffer tests
+
+if BUILD_XTENSA
+# these tests should be able to run on host after some tuning
 
 check_PROGRAMS += buffer_new
 buffer_new_SOURCES = src/audio/buffer/buffer_new.c src/audio/buffer/mock.c
@@ -213,6 +223,8 @@ buffer_copy_SOURCES += 	../../src/audio/component.c \
 buffer_copy_LDADD =  ../../src/host/libtb_common.a $(LDADD) -ldl
 else
 buffer_copy_LDADD =  ../../src/audio/libaudio.a $(LDADD)
+endif
+
 endif
 
 # component tests

